src/createServer.js

```diff
@@ -1,8 +1,142 @@
 'use strict';
 
+const http = require('node:http');
+const fs = require('node:fs');
+const path = require('node:path');
+const zlib = require('node:zlib');
+const formidable = require('formidable');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const compressionTypes = {
+    gzip: 'gzip',
+    deflate: 'deflate',
+    br: 'br',
+  };
+
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (url.pathname === '/' && req.method === 'GET') {
+      res.statusCode = 200;
+      res.setHeader('Content-Type', 'text/html');
+
+      const htmlPage = fs.createReadStream(
+        path.resolve('src/public/', 'index.html'),
+      );
+
+      htmlPage.pipe(res);
+
+      return;
+    }
+
+    if (url.pathname === '/compress' && req.method === 'GET') {
+      res.statusCode = 400;
+      res.end('Invalid method.');
+
+      return;
+    }
+
+    if (url.pathname === '/compress' && req.method === 'POST') {
+      const form = new formidable.IncomingForm();
+
+      form.parse(req, (err, fields, files) => {
+        if (err) {
+          res.statusCode = 400;
+          res.end('Invalid form data');
+
+          return;
+        }
+
+        const compressionType = fields.compressionType;
+
+        if (
+          !compressionType ||
+          !Object.keys(compressionTypes).includes(compressionType[0])
```

Comment:
The `compressionType` is being accessed as an array (`compressionType[0]`), but it is likely a string. Ensure that `compressionType` is correctly handled as a string, or verify that it is indeed an array and adjust the logic accordingly.